Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -1111,6 +1112,56 @@ Instrument cache.
     TS_ASSERT_THROWS(tester.openfullH5Path(transformationsPath),
                      H5::GroupIException &)
   }
+
+  void test_homogeneous_cylindrical_instrument_vertices_written_to_file() {
+
+    /*
+     test scenario: A test instrument with cylindrical homogeneous pixel shapes
+     is passed into SaveInstrument to be saved. Written to file are the three
+     coordinates of the vertices which describe the cylinder: the centre of the
+     base along the cylinder axis, the centre of the top face along the cylinder
+     axis, and the edge of the cylinder from the top face that is orthogonal to
+     the axis. The hieght, radius and base centre are known upon creation of the
+     test instrument. This test will verify that the vertex coordinates in file
+     are concurrent with the known height, radius, and base position."
+    */
+
+    // create RAII file resource for testing
+    ScopedFileHandle fileResource("test_cylindrical.hdf5");
+    std::string destinationFile = fileResource.fullPath();
+
+    auto cylindricalInstrument = ComponentCreationHelper::
+        createCylInstrumentWithVerticalOffsetsSpecified(
+            2, std::vector<double>{1, 2, 3}, 2, 1, 2, 1, 2);
+    auto instr = Mantid::Geometry::InstrumentVisitor::makeWrappers(
+        *cylindricalInstrument);
+
+    double cylinderHeight = 0.50;
+    double cylinderRadius = 0.25;
+
+    NexusGeometrySave::saveInstrument(instr, destinationFile,
+                                      DEFAULT_ROOT_PATH);
+    FullNXPath path{DEFAULT_ROOT_PATH, "instrument_with_tubes", "sixteenpack",
+                    PIXEL_SHAPE};
+
+    NexusFileReader tester(destinationFile);
+    double buffer[3][3];
+    tester.readDataSet3x3d(buffer, path, VERTICES);
+
+    Eigen::Vector3d base{buffer[0][0], buffer[0][1], buffer[0][2]};
+    Eigen::Vector3d top{buffer[1][0], buffer[1][1], buffer[1][2]};
+    Eigen::Vector3d edge{buffer[2][0], buffer[2][1], buffer[2][2]};
+
+    double radiusInFile = std::abs((top - edge).norm());
+    double heightInFile = std::abs((base - top).norm());
+
+    TS_ASSERT(std::abs(radiusInFile - cylinderRadius) <
```

Comment:
Use `TS_ASSERT_DELTA` or equivalent as gives better output info upon failure for tests of this type.

Reply:
Apart from that looks to be working OK.

Reply:
done